From 776b9d312d4749a7c679bed3c314c1dd763305cf Mon Sep 17 00:00:00 2001
From: Hui Min Mina Chou <minachou@andestech.com>
Date: Mon, 27 May 2024 17:27:06 +0800
Subject: [PATCH 008/168] riscv: andes: defconfig: add ltp.config to support
 full LTP testing

To increase test coverage, enable more kernel config for full LTP testing.
Add the following defconfig, planned for testing in the Jenkins CICD LTP
testsuite.
 * ae350_rv32_up_ltp_defconfig
 * ae350_rv32_smp_ltp_defconfig
 * ae350_rv64_up_ltp_defconfig
 * ae350_rv64_smp_ltp_defconfig

Signed-off-by: Hui Min Mina Chou <minachou@andestech.com>
---
 arch/riscv/Makefile           |  16 ++++
 arch/riscv/configs/ltp.config | 146 ++++++++++++++++++++++++++++++++++
 2 files changed, 162 insertions(+)
 create mode 100644 arch/riscv/configs/ltp.config

diff --git a/arch/riscv/Makefile b/arch/riscv/Makefile
index 4e2d92f1884d..b446456436e8 100644
--- a/arch/riscv/Makefile
+++ b/arch/riscv/Makefile
@@ -216,3 +216,19 @@ ae350_rv64_smp_sv48_defconfig:
 PHONY += ae350_rv64_ax25_amp_defconfig
 ae350_rv64_ax25_amp_defconfig:
 	$(Q)$(MAKE) -f $(srctree)/Makefile andes_defconfig andes-support.config 64-bit.config amp.config sv39.config
+
+PHONY += ae350_rv32_up_ltp_defconfig
+ae350_rv32_up_ltp_defconfig:
+	$(Q)$(MAKE) -f $(srctree)/Makefile andes_defconfig andes-support.config 32-bit.config ltp.config
+
+PHONY += ae350_rv32_smp_ltp_defconfig
+ae350_rv32_smp_ltp_defconfig:
+	$(Q)$(MAKE) -f $(srctree)/Makefile andes_defconfig andes-support.config 32-bit.config smp.config ltp.config
+
+PHONY += ae350_rv64_up_ltp_defconfig
+ae350_rv64_up_ltp_defconfig:
+	$(Q)$(MAKE) -f $(srctree)/Makefile andes_defconfig andes-support.config 64-bit.config sv39.config ltp.config
+
+PHONY += ae350_rv64_smp_ltp_defconfig
+ae350_rv64_smp_ltp_defconfig:
+	$(Q)$(MAKE) -f $(srctree)/Makefile andes_defconfig andes-support.config 64-bit.config smp.config sv39.config ltp.config
diff --git a/arch/riscv/configs/ltp.config b/arch/riscv/configs/ltp.config
new file mode 100644
index 000000000000..5f1f2cb4c743
--- /dev/null
+++ b/arch/riscv/configs/ltp.config
@@ -0,0 +1,146 @@
+CONFIG_CGROUPS=y
+CONFIG_CPUSETS=y
+CONFIG_NUMA=y
+CONFIG_KSM=y
+CONFIG_MEMCG=y
+CONFIG_SQUASHFS=y
+CONFIG_SQUASHFS_XZ=y
+CONFIG_XFS_FS=y
+CONFIG_XFS_QUOTA=y
+CONFIG_BTRFS_FS=y
+CONFIG_QUOTA=y
+CONFIG_QFMT_V2=y
+CONFIG_NFSD=y
+CONFIG_NFSD_V3_ACL=y
+CONFIG_NFSD_V4=y
+CONFIG_NFS_DEBUG=y
+CONFIG_NFSD_BLOCKLAYOUT=y
+CONFIG_NFSD_SCSILAYOUT=y
+CONFIG_NFSD_FLEXFILELAYOUT=y
+CONFIG_BINFMT_MISC=m
+CONFIG_ZSMALLOC=y
+CONFIG_ZRAM=m
+CONFIG_CRYPTO_LZO=y
+CONFIG_SWAP=y
+CONFIG_CAN=y
+CONFIG_CAN_DEV=m
+CONFIG_CAN_VCAN=m
+CONFIG_INPUT_MISC=y
+CONFIG_INPUT_UINPUT=m
+CONFIG_USER_DECRYPTED_DATA=y
+CONFIG_ROMFS_BACKED_BY_BLOCK=y
+CONFIG_MISC_FILESYSTEMS=y
+CONFIG_ROMFS_FS=y
+CONFIG_BLK_DEV_RAM=y
+CONFIG_TLS=y
+CONFIG_NF_CONNTRACK=y
+CONFIG_NETFILTER=y
+CONFIG_NETFILTER_XTABLES=y
+CONFIG_NETFILTER_XT_MATCH_STATE=y
+CONFIG_IP_NF_IPTABLES=y
+CONFIG_IP_NF_FILTER=y
+CONFIG_IP_NF_TARGET_REJECT=y
+CONFIG_TIME_NS=y
+CONFIG_NF_TABLES=y
+CONFIG_BLK_DEV_LOOP=y
+CONFIG_TUN=y
+CONFIG_OVERLAY_FS=y
+CONFIG_OVERLAY_FS_REDIRECT_DIR=y
+CONFIG_VETH=y
+CONFIG_CGROUP_NET_PRIO=y
+CONFIG_NET_CLS=y
+CONFIG_NET_CLS_BASIC=y
+CONFIG_NET_CLS_ACT=y
+CONFIG_NET_SCHED=y
+CONFIG_NET_CLS_CGROUP=y
+CONFIG_SCHED_AUTOGROUP=y
+CONFIG_FS_VERITY=y
+CONFIG_VSOCKETS_LOOPBACK=y
+CONFIG_TRANSPARENT_HUGEPAGE_MADVISE=y
+CONFIG_ARCH_ENABLE_THP_MIGRATION=y
+CONFIG_TRANSPARENT_HUGEPAGE=y
+CONFIG_CAN_SLCAN=y
+CONFIG_N_GSM=y
+CONFIG_SLIP=y
+CONFIG_PPP=y
+CONFIG_PPP_ASYNC=y
+CONFIG_MKISS=y
+CONFIG_HAMRADIO=y
+CONFIG_AX25=y
+CONFIG_CAIF=y
+CONFIG_CAIF_TTY=y
+CONFIG_PPS_CLIENT_LDISC=y
+CONFIG_PPP_SYNC_TTY=y
+CONFIG_HDLC=y
+CONFIG_N_HDLC=y
+CONFIG_CRYPTO_USER_API=y
+CONFIG_CRYPTO_USER_API_HASH=y
+CONFIG_CRYPTO_USER_API_SKCIPHER=y
+CONFIG_CRYPTO_USER_API_RNG=y
+CONFIG_CRYPTO_USER_API_AEAD=y
+CONFIG_CRYPTO_SM3_GENERIC=y
+CONFIG_CRYPTO_SHA3=y
+CONFIG_CRYPTO_SHA256=y
+CONFIG_CRYPTO_SHA512=y
+CONFIG_CRYPTO_CHACHA20=y
+CONFIG_CRYPTO_CHACHA20POLY1305=y
+CONFIG_CRYPTO_VMAC=y
+CONFIG_CRYPTO_SM4_GENERIC=y
+CONFIG_CRYPTO_AUTHENC=y
+CONFIG_CRYPTO_XCBC=y
+CONFIG_CRYPTO_SHA1=y
+CONFIG_BSD_PROCESS_ACCT=y
+CONFIG_DUMMY=m
+CONFIG_FANOTIFY=y
+CONFIG_SECURITY=y
+CONFIG_FANOTIFY_ACCESS_PERMISSIONS=y
+CONFIG_IP_SCTP=y
+CONFIG_NF_SOCKET_IPV6=y
+CONFIG_CRYPTO_USER=y
+CONFIG_CRYPTO_STATS=y
+# CONFIG_BPF_UNPRIV_DEFAULT_OFF is not set
+CONFIG_CRYPTO_PCRYPT=y
+CONFIG_BPFILTER=y
+CONFIG_CONNECTOR=y
+CONFIG_PROC_EVENTS=y
+# CONFIG_LOCK_TORTURE_TEST is not set
+# CONFIG_RCU_TORTURE_TEST is not set
+CONFIG_SND_SEQUENCER=y
+CONFIG_RDS=y
+CONFIG_TASKSTATS=y
+CONFIG_TASK_XACCT=y
+CONFIG_TASK_IO_ACCOUNTING=y
+CONFIG_XFRM_USER=y
+CONFIG_NF_TABLES_IPV4=y
+CONFIG_MEDIA_SUPPORT=y
+CONFIG_V4L_TEST_DRIVERS=y
+CONFIG_VIDEO_DEV=y
+CONFIG_VIDEO_V4L2=y
+CONFIG_MEDIA_CAMERA_SUPPORT=y
+CONFIG_BLK_CGROUP=y
+CONFIG_CGROUP_FAVOR_DYNMODS=y
+CONFIG_CFS_BANDWIDTH=y
+CONFIG_CGROUP_PIDS=y
+CONFIG_CGROUP_RDMA=y
+CONFIG_CGROUP_FREEZER=y
+CONFIG_CGROUP_HUGETLB=y
+CONFIG_PROC_PID_CPUSET=y
+CONFIG_CGROUP_DEVICE=y
+CONFIG_CGROUP_CPUACCT=y
+CONFIG_CGROUP_PERF=y
+CONFIG_CGROUP_BPF=y
+CONFIG_CGROUP_MISC=y
+CONFIG_CGROUP_DEBUG=y
+CONFIG_FS_ENCRYPTION=y
+CONFIG_QUOTACTL=y
+CONFIG_CRYPTO_CBC=y
+CONFIG_ZSWAP=y
+CONFIG_WAN=y
+CONFIG_QFMT_V1=y
+CONFIG_ENCRYPTED_KEYS=y
+CONFIG_SERIAL_NONSTANDARD=y
+CONFIG_TTY=y
+CONFIG_DNS_RESOLVER=m
+CONFIG_WATCH_QUEUE=y
+# CONFIG_MEDIA_CEC_SUPPORT is not set
+CONFIG_VIDEO_VIM2M=y
-- 
2.34.1

